src/features/machines/depositMachine.ts

```diff
@@ -1,178 +1,127 @@
-  context: {
-    blockchain: null,
-    depositAddress: null,
-    depositAsset: null,
-    depositAmount: null,
+  /** @xstate-layout N4IgpgJg5mDOIC5QTABwPawJYBcB02UAdlkVAGpYCGAcmFQE4FbGlQDEE6RYepAbugDWvFBmz5CJMpVr0mUtggHoAxlRxZuAbQAMAXT37Eocbi1ETIAB6IAjAA4AzHgBsAVgCcd9wCZfngDsrq52ACy+ADQgAJ72Ybp4Tp5OYcGeYQ66DnZ2AL550WKYuMysMtR0jGXSHGAMDOhMqAA2GgBmTQC2eMUSNWyyVQostcpEguqaOgZGVmbTlkg2iAC0ri5hYZ66Eb5eaT7RcQjhicmp3oGB7gmZBUVoJZKjg5XyePz1WO0xbOxzZYLCxWWwIfzuPC+JyBXw5XS+Vy6XLQ46IJyuTx4OxOXS6JzuBGuba6TwPEB9UqKCpyapfBg-P5kAF2YxA54g5ZgtIuQnJLa6EKEsIEtEIDFYnF4glEklkwoUp79akUd7VABGjSoEHUsE0zK4PD4E2EoiVVNeNOGeE16G1uv1UHGkw0FiMgNMHO4oMQvl2eG8uIcAXCSP8DjF1yxhIcrhywccGXljwWeBgPAYroN3F4KhEvXN+HT9SzTpUUzdswM8y9S1AYMCnlcSWuiM8bc8DmCYtWBLwIqbgQcEUJvmuDgKCqI6BQ8GWlJwNYknPrax8WK2Oz2B0CR1iazCdkSCKCTec7lx7kC5IXAyt8iX5m9XLXuX7JO3GV37h7ft5ASHfxPGA3wRRvQs71VWkRnKKBH0WH0EDCZtAgxXIuzha4cSifdwQDQIEVJHYh0Jbx3HA1MVSGD56UZNh4JXFZwSvbFoWAkIbiCYkxVAsIAzxQJkLxTs0jSCjnkg6jqgAQXVJocEgBjn1XBB1jcUDg0DQTdyRG4xSCXx3xxLZ3FMv1h3E5VLSg61bXtKg9Xo9ll2Upi7BCfCMg8LtdF3BExQvQyjycRE0lM3FG0si1YKkpgaHQHAAAIADF0AAVyICBEqaRKAEkJioFosAgJS6yYkVIUcDFoTxVwbnc-TYSM1IwlMvxsjCKKXhitUmAAYXQLpWjABSSucp8yrBNi8GcYI0gwsMcUawytmM1qzI6rq0zADNS1KxDBIcFtYUxDsu1cX9kX7ZI6s8Pw-BSJwtuLTNHTwABxHb6hGxTxoQl8EGHOwZscYkghuVDBICyEkTOOxvEPMLr0nIA */
+  context: ({ input }: { input: Input }) => {
+    return {
+      depositRef: null,
+    }
   },
-  id: "deposit_machine",
-  initial: "Idle",
+  id: "deposit",
+  type: "parallel",
```

Comment:
If I understand correctly, these aren't parallel states. You either follow the Near path or the Passive deposit path (depositing to the generated address).

Reply:
I see several reasons to do this in different threads:
1. These states have different inputs and outputs
2. When adding X path (Ethereum or other) we won't have to make changes to the current threads